Src/Sarif.PatternMatcher/FlexMatchValueComparer.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+
+using Microsoft.RE2.Managed;
+
+namespace Microsoft.CodeAnalysis.Sarif.PatternMatcher
+{
+    internal class FlexMatchValueComparer : IEqualityComparer<FlexMatch>
```

Comment:
>FlexMatchValueComparer [](http://example.com/codeflow?start=19&length=22)

This should have come in with some unit tests, feel free to add a couple, its not much work! :)

Reply:
I did two major changes:
1. moved this to RE2, so people can use it and this would not depend on SarifPatternMatcher
2. added tests